dashboard/app/controllers/home_controller.rb

```diff
@@ -85,11 +85,32 @@ def init_homepage
         current_user.gallery_activities.order(id: :desc).page(params[:page]).per(GALLERY_PER_PAGE)
       @force_race_interstitial = params[:forceRaceInterstitial]
       @force_school_info_interstitial = params[:forceSchoolInfoInterstitial]
-      @recent_courses = current_user.recent_courses_and_scripts.slice(0, 2)
+      @recent_courses = current_user.recent_courses_and_scripts
+
+      unless current_user.teacher?
+        @recent_courses = current_user.recent_courses_and_scripts.drop(1)
+      end
 
       if current_user.teacher?
         @sections = current_user.sections.map(&:summarize)
       end
+
+      unless current_user.teacher?
+        script = current_user.primary_script
+        if script
+          script_level = current_user.next_unpassed_progression_level(script)
+        end
+
+        if script && script_level
+          @student_top_course = {
+            assignableName: data_t_suffix('script.name', script[:name], 'title'),
+            lessonName: script_level.stage.localized_title,
+            linkToOverview: script_path(script),
+            linkToLesson: "Ask Brent?"
+            #build_script_level_path(script_level)
+          }
```

Comment:
What is the nature of the not working? Is it just that the method is not in scope to you here? If so, you might be able to just include LevelsHelper (where that is defined).